src/librawspeed/decoders/DngDecoderSlices.cpp

```diff
@@ -158,10 +159,9 @@ void DngDecoderSlices::decodeSlice(DngDecoderThread* t) {
     while (!t->slices.empty()) {
       DngSliceElement e = t->slices.front();
       t->slices.pop();
-      LJpegPlain l(*mFile, e.byteOffset, e.byteCount, mRaw);
-      l.mDNGCompatible = mFixLjpeg;
+      LJpegDecompressor d(*mFile, e.byteOffset, e.byteCount, mRaw);
```

Comment:
Glad you like it. As I said: I'd like to get this off my plate. Once this is merged, I'll wrap up the other open pr and maybe 1-2 additional 'loose ends' from the work done earlier.